warrior/Framework/Utils/data_Utils.py

```diff
@@ -900,15 +897,29 @@ def validate():
     result, err_msg, exp = validate()
     keys = key.split('.')
     value = get_object_from_datarepository(keys[0])
-    for k in keys[1:]:
-        value = value[k]
-    if not value:
-        err_msg += "key {} not present in data repository\n".format(key)
+    key_err_msg = "key {} not present in data repository\n"
+    if value:
+        try:
+            for k in keys[1:]:
+                value = value[k]
+            if result == "ERROR" or result == "EXCEPTION":
+                print_error(err_msg)
+            elif not comp_funcs[comparison](value, exp):
+                result = "FALSE"
+                print_warning("The key, value pair '{0}:{1}' present in the  "
+                              "data_repository doesn't satisfy the expected value & "
+                              "condition '{2}:{3}'".format(key, value, comparison, expected))
+            else:
+                print_info("The key, value pair '{0}:{1}' present in the  "
+                           "data_repository satisfies the expected value & condition "
+                           "'{2}:{3}'".format(key, value, comparison, expected))
+        except KeyError:
+            print_error(key_err_msg.format(key))
```

Comment:
Hi @sugan19,

Please catch a general exception and print it out here. You can refer to line 879 of this file.

Reply:
@sanikakulkarni Addressed your comments